### DISCUSSION TITLE
Running Shell Scripts with Real-Time Output in a Text Widget

### DISCUSSION THREAD

**Discussion by latushko1** [Textualize/textual#5393] [Q&A] [answered]
Hello,

I'm working on a project where I need to run shell scripts and display their output in a text widget in real time. I tried using subprocess but couldn't achieve the desired result. (I have python 3.7 in my working machine 😭)

Is there a recommended way to execute shell scripts and stream their output to a text widget with minimal latency while ensuring safety and stability? Any guidance or examples would be greatly appreciated.

Thank you!

**Comment by willmcgugan** [comment, marked as answer]
Subprocess would be the way to go. You should do that in a worker to avoid preventing the UI from updating.

I'm afraid we dropped 3.7 support in Textual a while back. So you may be stuck at an older Textual.

**Reply by latushko1** [reply]
Indeed, the worker helped achieve the desired result, thank you for your help.
